internal/kubernetes/operator/features/crds.go

```diff
@@ -0,0 +1,203 @@
+package features
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+	"time"
+
+	apiextensions "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	"sigs.k8s.io/yaml"
+)
+
+const (
+	urlTemplate                 = "https://raw.githubusercontent.com/mongodb/mongodb-atlas-kubernetes/v%s/bundle/manifests/atlas.mongodb.com_%s.yaml"
+	maxDepth                    = 100
+	requestTimeout              = 10 * time.Second
+	ResourceVersion             = "app.kubernetes.io/version"
+	ResourceAtlasProject        = "atlasprojects"
+	ResourceAtlasDeployment     = "atlasdeployments"
+	ResourceAtlasDatabaseUser   = "atlasdatabaseusers"
+	ResourceAtlasBackupSchedule = "atlasbackupschedules"
+	ResourceAtlasBackupPolicy   = "atlasbackuppolicies"
+	ResourceAtlasTeam           = "atlasteams"
+)
+
+var (
+	ErrVersionNotSupportedFmt    = "version '%s' is not supported"
+	ErrDownloadResourceFailedFmt = "can not download resource '%s': %v"
+	ErrDocumentIsEmpty           = errors.New("document is empty")
+	ErrDocumentHasNoVersions     = errors.New("document contains no versions")
+	ErrDocumentHasNoSchema       = errors.New("document contains no Schema")
+	ErrDocumentHasNoSpec         = errors.New("document contains no Spec")
+
+	VersionsToResourcesMap = map[string][]string{
+		"1.4.1": {
+			ResourceAtlasProject,
+			ResourceAtlasDeployment,
+			ResourceAtlasDatabaseUser,
+			ResourceAtlasBackupSchedule,
+			ResourceAtlasBackupPolicy,
+		},
+		"1.5.0": {
+			ResourceAtlasDatabaseUser,
+			ResourceAtlasProject,
+			ResourceAtlasDeployment,
+			ResourceAtlasBackupSchedule,
+			ResourceAtlasBackupPolicy,
+			ResourceAtlasTeam,
+		},
+		"1.5.1": {
+			ResourceAtlasDatabaseUser,
+			ResourceAtlasProject,
+			ResourceAtlasDeployment,
+			ResourceAtlasBackupSchedule,
+			ResourceAtlasBackupPolicy,
+			ResourceAtlasTeam,
+		},
+	}
+)
+
+func GetSupportedVersions() []string {
+	result := make([]string, 0, len(VersionsToResourcesMap))
+	for version := range VersionsToResourcesMap {
+		result = append(result, version)
+	}
+	return result
+}
+
+func downloadResource(client *http.Client, resourceName string, version string) (*apiextensions.CustomResourceDefinition, error) {
+	ctx, cancelF := context.WithTimeout(context.Background(), requestTimeout)
+	defer cancelF()
+
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, fmt.Sprintf(urlTemplate, version, resourceName), nil)
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := client.Do(req)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	data, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return nil, err
+	}
+
+	decoded := &apiextensions.CustomResourceDefinition{}
+	err = yaml.Unmarshal(data, decoded)
+	if err != nil {
+		return nil, err
+	}
+
+	return decoded, nil
+}
```

Comment:
cloud/should we wrap this in the store layer?

Reply:
I thought about that, but the store seems be working with the Atlas API. I can do that, but I'm not sure if that is needed.

Reply:
When I designed the CLI I'd did with 3 core layers, the config layer, one the settings, the store layer, or how to access data and the CLI layer, the thing that flies them (and other parts) together.
You can see store as a transparent way to access data (DAO?) In the CLI no matter where the data is, proof of this model, the store is able to handle atlas or ops manager in a transparent way for consumers. This approach let's us have some decoupling between the biz logic and data and why I'm asking here if long term you can benefit of the decoupling both

Reply:
Yes, I noticed this approach, I just didn't want to extend it with this small feature, as it has nothing to do with the Atlas API. But okay, let me add it to the `Store`.

Reply:
Fixed by moving CRDs download logic into a separate provider